File: src/cpp/FizzBuzz.c
// ヘッダファイルの取り込み
#include<stdio.h> 

// 最初に実行されるmain関数
int main(void) {
  int i; // i++と++i 前置後置インクリメント
  for(i = 1; i <= 100; ++i) {
    if (i % 3 == 0 && i % 5 == 0) { 
      printf("FizzBuzz");
    } else if (i % 3 == 0) {
      printf("Fizz");
    } else if (i % 5 == 0) {
      printf("Buzz");
    } else {
      printf("%d", i);
    }
    printf("\n");
  }
  return 0;
}

File: src/fizzbuzz.ml
(* FizzBuzzを返す関数を定義 --- [1] *)
let fizzbuzz i = match i mod 3, i mod 5 with
      0, 0 -> "FizzBuzz"
    | 0, _ -> "Fizz"
    | _, 0 -> "Buzz"
    | _    -> string_of_int i

(* 100回、fizzbuzzを呼び出す --- [2] *)
let () =
  for i = 1 to 100 do print_endline @@ fizzbuzz i done

